tests/integration/apigateway/test_apigateway_integrations.py

```diff
@@ -53,6 +56,112 @@ def test_http_integration(create_rest_apigw, aws_client, echo_http_server):
     assert response.status_code == 200
 
 
+@pytest.fixture
+def status_code_http_server(httpserver: HTTPServer):
+    """Spins up a local HTTP echo server and returns the endpoint URL"""
+    if is_aws_cloud():
+        return f"{PUBLIC_HTTP_ECHO_SERVER_URL}/"
+
+    def _echo(request: Request) -> Response:
+        result = {
+            "data": request.data or "{}",
+            "headers": dict(request.headers),
+            "url": request.url,
+            "method": request.method,
+        }
+        status_code = request.url.rpartition("/")[2]
+        response_body = json.dumps(result)
+        print(result)
+        return Response(response_body, status=int(status_code))
+
+    httpserver.expect_request("").respond_with_handler(_echo)
+    http_endpoint = httpserver.url_for("/")
+    return http_endpoint
+
+
+@pytest.mark.aws_validated
+def test_http_integration_status_code_selection(
+    create_rest_apigw, aws_client, status_code_http_server
+):
+    api_id, _, root_id = create_rest_apigw(name="my_api", description="this is my api")
```

Comment:
Are we adding the policy tests on another PR? I know this took you from what you initially intended to do, so just checking

Reply:
Yes, it is in the linked -ext PR.